src/stores/EventsStore.tsx

```diff
@@ -1,24 +1,60 @@
+// EventsStore.ts
+
 import { makeAutoObservable } from "mobx"
 import { IEvent } from "../firebase/interfaces"
+import { validateEvent } from "../firebase/validation/event"
+import { ValidationErrorItem, ValidationResult } from "joi"
 
 export class EventsStore {
     dbEvents: IEvent[] = []
     event: Partial<IEvent> = {}
+    validationResult: ValidationResult | undefined = undefined
+    modifiedFields: Set<keyof IEvent> = new Set()
 
     constructor() {
         makeAutoObservable(this)
+        this.validationResult = validateEvent({} as Partial<IEvent>)
+        this.createEvent = this.createEvent.bind(this)
+        this.getErrorForKey = this.getErrorForKey.bind(this)
     }
 
     setEvent<K extends keyof IEvent, V extends IEvent[K]>(key: K, value: V) {
         this.event[key] = value
+        this.modifiedFields.add(key)
```

Comment:
To ni vredi. TextField komponenta se mora sama zavedat kdaj se je nekaj spremenilo ali ne. To pomeni da naredi rešitev samo znotraj TextFielda komponente.

Reply:
Se pravi "modifieldFields" odpade? V to glih nisn bil ziher... Pa nisn našo druge rešitve... Premislim še malo pa popravim...

Reply:
V textfield si daj nek state showError, ki je v štartu false. Na input change ali input blur, spremeniš showError na true.